app/javascript/packages/document-capture/components/file-input.jsx

```diff
@@ -193,6 +197,7 @@ function FileInput(props, ref) {
   }
 
   const shownErrorMessage = errorMessage ?? ownErrorMessage;
+  const hasEffectiveValue = value && !isValuePending;
```

Comment:
The current build failure is due to TypeScript not following this new assigned value as guaranteeing that `value` would be truthy within the `&&` conditions below.

```
app/javascript/packages/document-capture/components/file-input.jsx:271:41 - error TS2345: Argument of type 'string | Blob | null | undefined' is not assignable to parameter of type 'string | Blob'.
  Type 'undefined' is not assignable to type 'string | Blob'.

271           {hasEffectiveValue && isImage(value) && (
                                            ~~~~~

app/javascript/packages/document-capture/components/file-input.jsx:276:22 - error TS2322: Type 'string | null | undefined' is not assignable to type 'string | undefined'.
  Type 'null' is not assignable to type 'string | undefined'.

276                 <img src={value} alt="" className="usa-file-input__preview-image" />
```

On a hunch, I tried seeing if upgrading TypeScript could help, and it does seem like newer versions are better at understanding this. However, there's a few other (unrelated) errors that need to be resolved to that upgrade.

I'm on the fence between:

1. Get rid of this assignment and inline `value && !isValuePending` where we're using it below
2. Separately upgrade TypeScript, then rebase this branch

Reply:
could we just `/* type ... */` cast it or have `isImage` take in `any` for now?

Reply:
> could we just `/* type ... */` cast it or have `isImage` take in `any` for now?

We could, yeah. Personally I'd be more in favor of duplicating the logic, since we're really only using it in a few places.

I started to see what it might take to upgrade TypeScript, but it looks like we're partially blocked by some upstream dependencies not yet being compatible.